adapters/celestia/src/verifier/mod.rs

```diff
@@ -182,20 +184,19 @@ impl da::DaVerifier for CelestiaVerifier {
 
         // Check the validity and completeness of the rollup row proofs, against the DAH.
         // Extract the data from the row proofs and build a namespace_group from it
-        let rollup_shares_u8 = self.verify_row_proofs(completeness_proof, &block_header.dah)?;
-        if rollup_shares_u8.is_empty() {
+        let verified_shares = self.verify_row_proofs(completeness_proof, &block_header.dah)?;
+        if verified_shares.is_empty() {
             if txs.is_empty() {
                 return Ok(validity_condition);
             }
             return Err(ValidationError::MissingTx);
         }
-        let namespace = NamespaceGroup::from_shares_unchecked(rollup_shares_u8);
 
         // Check the e-tx proofs...
         // TODO(@preston-evans98): Remove this logic if Celestia adds blob.sender metadata directly into blob
         let mut tx_iter = txs.iter();
         let square_size = block_header.dah.row_roots.len();
-        for (blob, tx_proof) in namespace.blobs().zip(inclusion_proof.into_iter()) {
+        for (blob, tx_proof) in verified_shares.blobs().zip(inclusion_proof.into_iter()) {
```

Comment:
This doesn't look like a new bug - but I think this check is broken. If `inclusion_proof` is an empty iterator, this loop will never execute. Looks like we need to check that `inclusion_proof.len()` == `verified_shares.blobs().len()`

Reply:
this case should be covered now

Reply:
https://github.com/Sovereign-Labs/sovereign-sdk/pull/948/commits/d755a2d3a2a6b15da77ea82abdb553da554ffcc0

Reply:
Fanstastic, thanks @zvolin!